src/expr/src/relation/mod.rs

```diff
@@ -2769,6 +2769,18 @@ impl RowSetFinishing {
             })
             .unwrap_or(limit);
 
+        // Bail early if we know our results won't fit into our Vec
+        //
+        // Note: this check is a _lower bound_ as to how much memory
+        // we'll need for results. Calculating the actual answer would
+        // require us to iterate through all of the rows
+        if return_size.saturating_mul(std::mem::size_of::<Row>()) >= max_result_size {
```

Comment:
Small nit, below we use `>` instead of `>=`.

> `if total_bytes > max_result_size {`

Can we use `>` here too? I think you'll have to switch to `checked_mul` to handle the rare case of `max_result_size == usize::MAX`.

Also while you're in here, I think the `total_bytes += new_row.byte_len();` can also overflow. Would you mind updating that one too.

Reply:
updated this logic to use just `>`!

as discussed offline, we're okay with not checking for overflow of `total_bytes += new_row.byte_len()`